### PR TITLE
Added detail view

### PR DESCRIPTION
**Pull request by sleonardo**
## Summary ##

**Trello cards**:
> https://trello.com/c/hZEnajnw
> https://trello.com/c/PcscLEga
> https://trello.com/c/TkdhHZsk
> https://trello.com/c/pLEv8r8l

View for the book detail was created.

### iPhone 8

Captura de Pantalla 2021-11-12 a la(s) 11 08 48 a  m

### iPhone X

Captura de Pantalla 2021-11-12 a la(s) 11 08 48 a  m

Captura de Pantalla 2021-11-12 a la(s) 11 08 48 a  m

st ##

## Screenca

> detail-book-view-2

> detail-book-view

### CODE REVIEW

File: WBooks/App/BookDetail/BookDetailViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+//
+//  BookDetailViewModel.swift
+//  WBooks
+//
+//  Created by leonardo.a.simoza on 09/11/21.
+//
+
+import Foundation
```
Comment (JuanJSilva) [inline WBooks/App/BookDetail/BookDetailViewModel.swift:8]: It's necessary this import?

File: WBooks/App/BookDetail/BookDetailViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+protocol BookDetailViewModelType {
+    var navBarTitle: String { get }
+    func createInfographicsViewModel() -> InfographicViewModel
+}
+
+public class BookDetailViewModel: BookDetailViewModelType {
+
```
Comment (JuanJSilva) [inline WBooks/App/BookDetail/BookDetailViewModel.swift:16]: Remove extra line

File: WBooks/App/BookDetail/BookDetailViewModel.swift (hunk 1 above)
Comment (JuanJSilva) [reply WBooks/App/BookDetail/BookDetailViewModel.swift:8]: This is repeated in several files

File: WBooks/App/BookDetail/BookDetailViewModel.swift (hunk 2 above)
Comment (JuanJSilva) [reply WBooks/App/BookDetail/BookDetailViewModel.swift:16]: This is repeated in several files

File: WBooks/App/BookDetail/Comments/Cells/CommentCell.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    @IBOutlet weak var commentDescription: UILabel!
+    
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        backgroundColor = .clear
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+    }
```
Comment (NickMano) [inline WBooks/App/BookDetail/Comments/Cells/CommentCell.swift:25]: Do we need it?

File: WBooks/App/BookDetail/Infographic/InfographicView.swift [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        didSet {
+            addToWishlistButton.setStyle(.transparent)
+        }
+    }
+    @IBOutlet weak var rentButton: UIButton!
+    
+    // MARK: Custom methods
+    func configure(viewModel: InfographicViewModel) {
+        bookTitleLabel.text = viewModel.title
+        bookStatusLabel.text = viewModel.availability(viewModel.status)
```
Comment (NickMano) [inline WBooks/App/BookDetail/Infographic/InfographicView.swift:79]: Can we simplify the method availabilty?

File: WBooks/App/BookDetail/Infographic/InfographicView.swift [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        bookStatusLabel.textColor = viewModel.isBookAvailable ? UIColor.appColor(.availableColor) : UIColor.appColor(.unavailableColor)
```
Comment (NickMano) [inline WBooks/App/BookDetail/Infographic/InfographicView.swift:80]: I think is better generate a computed property like 
`viewModel.availableTextColor`

File: WBooks/App/BookDetail/Infographic/InfographicView.swift [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+                .transition(.fade(1)),
+                .cacheOriginalImage
+            ]) { result in
+            switch result {
+            case .success(let value):
+                DDLogDebug("Task done for: \(value.source.url?.absoluteString ?? "")")
+            case .failure(let error):
+                DDLogError("Job failed: \(error.localizedDescription)")
+            }
+        }
```
Comment (NickMano) [inline WBooks/App/BookDetail/Infographic/InfographicView.swift:98]: We can split this in a method to improve readaility

File: WBooks/App/BookDetail/Infographic/InfographicViewController.swift [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        infographicView.configure(viewModel: viewModel)
+    }
+    
+    override func loadView() {
+        super.loadView()
+        view = infographicView
+    }
+
```
Comment (NickMano) [inline WBooks/App/BookDetail/Infographic/InfographicViewController.swift:35]: Extra space

File: WBooks/Common/Util/Extensions/UIButton+Extension.swift [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+        self.layer.borderWidth = 1
+    }
+    
+    func defaultStyle() {
+        self.layer.cornerRadius = 22
+        self.setTitleColor(.white, for: .normal)
+        self.clipsToBounds = true
+    }
+    
+    func addGradient(colors: [CGColor], _ count: Int) {
```
Comment (NickMano) [inline WBooks/Common/Util/Extensions/UIButton+Extension.swift:52]: Can count param be the array count?

Comment (sleonardo) [reply WBooks/Common/Util/Extensions/UIButton+Extension.swift]: Yes, ready!

File: WBooks/App/BookDetail/Infographic/InfographicView.swift (hunk 4 above)
Comment (sleonardo) [reply WBooks/App/BookDetail/Infographic/InfographicView.swift]: Ready

File: WBooks/App/BookDetail/Infographic/InfographicView.swift (hunk 5 above)
Comment (sleonardo) [reply WBooks/App/BookDetail/Infographic/InfographicView.swift]: Ready!

File: WBooks/App/BookDetail/Comments/Cells/CommentCell.swift (hunk 3 above)
Comment (sleonardo) [reply WBooks/App/BookDetail/Comments/Cells/CommentCell.swift]: Ready

**Review by NickMano — APPROVED** [review]

**Review by JuanJSilva — APPROVED** [review]

**Pull request merged**